sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/TableAsyncClientTest.java

```diff
@@ -822,14 +824,24 @@ void submitTransactionAsyncWithSameRowKeys() {
             TableTransactionActionType.CREATE, new TableEntity(partitionKeyValue, rowKeyValue)));
 
         // Act & Assert
-        StepVerifier.create(tableClient.submitTransactionWithResponse(transactionalBatch))
-            .expectErrorMatches(e -> e instanceof TableTransactionFailedException
-                && e.getMessage().contains("An action within the operation failed")
-                && e.getMessage().contains("The failed operation was")
-                && e.getMessage().contains("CreateEntity")
-                && e.getMessage().contains("partitionKey='" + partitionKeyValue)
-                && e.getMessage().contains("rowKey='" + rowKeyValue))
-            .verify();
+        if (IS_COSMOS_TEST) {
+            StepVerifier.create(tableClient.submitTransactionWithResponse(transactionalBatch))
+                .expectErrorMatches(e -> e instanceof TableServiceException
+                    && e.getMessage().contains("Status code 400")
+                    && e.getMessage().contains("InvalidDuplicateRow")
+                    && e.getMessage().contains("The batch request contains multiple changes with same row key.")
+                    && e.getMessage().contains("An entity can appear only once in a batch request."))
+                .verify();
+        } else {
+            StepVerifier.create(tableClient.submitTransactionWithResponse(transactionalBatch))
+                .expectErrorMatches(e -> e instanceof TableTransactionFailedException
+                    && e.getMessage().contains("An action within the operation failed")
+                    && e.getMessage().contains("The failed operation was")
+                    && e.getMessage().contains("CreateEntity")
+                    && e.getMessage().contains("partitionKey='" + partitionKeyValue)
+                    && e.getMessage().contains("rowKey='" + rowKeyValue))
+                .verify();
+        }
```

Comment:
Because Cosmos is not consistent with Storage in this particular error scenario, it returns a `400` status code with the error message shown above instead of returning a `202` multipart response. I will talk to the Cosmos team about this and figure if it's intended or not.

Reply:
Either way, is this something that we can handle inside Tables SDK to map it to a common exception type for both Storage and Cosmos? It would be odd for users to get  `TableTransactionFailedException` and `TableServiceException` for the same transaction depending on the source service.

Reply:
I think there is something I can do as a workaround in the meantime to return the same type of exception, but I would prefer to wait and see if this is a service bug.